dds/idl/value_reader_generator.cpp

```diff
@@ -198,31 +199,30 @@ namespace {
           indent << "if (!value_reader.read_" << primitive_type(pt) << '(' << var_name << ")) return false;\n";
       }
     } else {
-      // TODO(tyler) Not sure how this affects the original
-      std::string value_expr = expression + accessor;
       if (!(c & CL_ENUM)) {
-        const std::string value_name = optional ? "tmp" : expression + accessor;
         const std::string type_name = scoped(type->name());
         switch (filter_kind) {
-        case FieldFilter_NestedKeyOnly:
-          value_expr = field_name + "_nested_key_only";
-          be_global->impl_ <<
-            indent << "const NestedKeyOnly<" << type_name << "> " <<
-            value_expr << "(" << value_name << ");\n";
-          break;
-        case FieldFilter_KeyOnly:
-          value_expr = field_name + "_key_only";
-          be_global->impl_ <<
-            indent << "const KeyOnly<" << type_name << "> " <<
-            value_expr << "(" << value_name << ");\n";
-          break;
+        case FieldFilter_NestedKeyOnly: {
+            const std::string value_expr = field_name + "_nested_key_only";
+            be_global->impl_ <<
+              indent << "const NestedKeyOnly<" << type_name << "> " <<
+              value_expr << "(" << var_name << ");\n";
+            break;
+          }
+        case FieldFilter_KeyOnly: {
+            const std::string value_expr = field_name + "_key_only";
+            be_global->impl_ <<
+              indent << "const KeyOnly<" << type_name << "> " <<
+              value_expr << "(" << var_name << ");\n";
+            break;
+          }
         default:
           break;
         }
       }
 
       be_global->impl_ <<
-        indent << "if (!vread(value_reader, " << value_expr <<
+        indent << "if (!vread(value_reader, " << var_name <<
```

Comment:
`var_name` should be `value_expr`.

Reply:
Sorry about that, I think the newest commit does what I wanted, for non keyonly optional structs they need to use the 'tmp' variable that was created earlier